host/xtest/acipher_perf.c

```diff
@@ -0,0 +1,921 @@
+	fprintf(stderr, " [-t main_algo] [-m mode] [-n LOOP] [-r|--no-inited] [-d WIDTH_BITS]");
+	fprintf(stderr, " [-k SIZE] [-a rsa_algo] [-s salt_len] [-v [-v]] [-w SEC]");
+	fprintf(stderr, "\n");
+	fprintf(stderr, "Asymmetric performance testing tool for OP-TEE\n");
+	fprintf(stderr, "\n");
+	fprintf(stderr, "Options:\n");
+	fprintf(stderr, "  -h|--help        Print this help and exit\n");
+	fprintf(stderr, "  -a ALGO          asymmetric algo\n");
+	fprintf(stderr, "                   DH, RSA_GENKEYPAIR, RSA_NOPAD_ENCRYPT, RSA_NOPAD_DECRYPT\n");
+	fprintf(stderr, "                   RSAES_PKCS1_V1_5_ENCRYPT, RSAES_PKCS1_V1_5_DECRYPT\n");
+	fprintf(stderr, "                   RSAES_PKCS1_OAEP_SHA1_ENCRYPT, RSAES_PKCS1_OAEP_SHA1_DECRYPT\n");
+	fprintf(stderr, "                   RSAES_PKCS1_OAEP_SHA224_ENCRYPT, RSAES_PKCS1_OAEP_SHA224_DECRYPT\n");
+	fprintf(stderr, "                   RSAES_PKCS1_OAEP_SHA256_ENCRYPT, RSAES_PKCS1_OAEP_SHA256_DECRYPT\n");
+	fprintf(stderr, "                   RSAES_PKCS1_OAEP_SHA384_ENCRYPT, RSAES_PKCS1_OAEP_SHA384_DECRYPT\n");
+	fprintf(stderr, "                   RSAES_PKCS1_OAEP_SHA512_ENCRYPT, RSAES_PKCS1_OAEP_SHA512_DECRYPT\n");
+	fprintf(stderr, "                   RSASSA_PKCS1_V1_5_SHA1_SIGN, RSASSA_PKCS1_V1_5_SHA1_VERIFY\n");
+	fprintf(stderr, "                   RSASSA_PKCS1_V1_5_SHA224_SIGN, RSASSA_PKCS1_V1_5_SHA224_VERIFY\n");
+	fprintf(stderr, "                   RSASSA_PKCS1_V1_5_SHA256_SIGN, RSASSA_PKCS1_V1_5_SHA256_VERIFY\n");
+	fprintf(stderr, "                   RSASSA_PKCS1_V1_5_SHA384_SIGN, RSASSA_PKCS1_V1_5_SHA384_VERIFY\n");
+	fprintf(stderr, "                   RSASSA_PKCS1_V1_5_SHA512_SIGN, RSASSA_PKCS1_V1_5_SHA512_VERIFY\n");
+	fprintf(stderr, "                   RSASSA_PKCS1_PSS_MGF1_SHA1_SIGN, RSASSA_PKCS1_PSS_MGF1_SHA1_VERIFY\n");
+	fprintf(stderr, "                   RSASSA_PKCS1_PSS_MGF1_SHA224_SIGN, RSASSA_PKCS1_PSS_MGF1_SHA224_VERIFY\n");
+	fprintf(stderr, "                   RSASSA_PKCS1_PSS_MGF1_SHA256_SIGN, RSASSA_PKCS1_PSS_MGF1_SHA256_VERIFY\n");
+	fprintf(stderr, "                   RSASSA_PKCS1_PSS_MGF1_SHA384_SIGN, RSASSA_PKCS1_PSS_MGF1_SHA384_VERIFY\n");
+	fprintf(stderr, "                   RSASSA_PKCS1_PSS_MGF1_SHA512_SIGN, RSASSA_PKCS1_PSS_MGF1_SHA512_VERIFY\n");
+	fprintf(stderr, "                   ECDSA_SIGN, ECDSA_VERIFY, ECDH, X25519\n");
+	fprintf(stderr, "  -l LOOP          Inner loop iterations [%u]\n", l);
+	fprintf(stderr, "  -n LOOP          Outer test loop iterations [%u]\n", n);
+	fprintf(stderr, "  -r|--random      Get input data from /dev/urandom (default: all zeros)\n");
+	fprintf(stderr, "  -k SIZE          Plaintext byte length [%u]\n", size);
+	fprintf(stderr, "  -d WIDTH_BITS    Private key size in bits [%u]\n", width_bits);
+	fprintf(stderr, "                   ECC: 192, 224, 256, 384 or 521, DH: <= 2048, RSA: <= 4096\n");
+	fprintf(stderr, "  -s SALT_LEN      Salt length in bytes (only when ALGO is one of RSA SSA_PKCS1_PSS_*)[%u]\n", salt_len);
```

Comment:
I got a the following when salt is not specified. The code needs to prevent that
```

root@imx8mn-ddr4-evk:~# xtest acipher-perf -n 10 -a RSASSA_PKCS1_PSS_MGF1_SHA256_VERIFY
E/TC:? 0
E/TC:? 0 TA panicked with code 0xffff0001
E/LD:  Status of TA 02a42f43-d8b7-4a57-aa4d-87bd9b5587cb
E/LD:   arch: aarch64
E/LD:  region  0: va 0x80005000 pa 0x56201000 size 0x002000 flags rw-s (ldelf)
E/LD:  region  1: va 0x80007000 pa 0x56203000 size 0x008000 flags r-xs (ldelf)
E/LD:  region  2: va 0x8000f000 pa 0x5620b000 size 0x001000 flags rw-s (ldelf)
E/LD:  region  3: va 0x80010000 pa 0x5620c000 size 0x004000 flags rw-s (ldelf)
E/LD:  region  4: va 0x80014000 pa 0x56210000 size 0x001000 flags r--s
E/LD:  region  5: va 0x80015000 pa 0x56236000 size 0x001000 flags rw-s (stack)
E/LD:  region  6: va 0x80016000 pa 0x4a6da000 size 0x001000 flags rw-- (param)
E/LD:  region  7: va 0x80017000 pa 0x4aa24000 size 0x001000 flags rw-- (param)
E/LD:  region  8: va 0x80021000 pa 0x00001000 size 0x019000 flags r-xs [0]
E/LD:  region  9: va 0x8003a000 pa 0x0001a000 size 0x00c000 flags rw-s [0]
E/LD:   [0] 02a42f43-d8b7-4a57-aa4d-87bd9b5587cb @ 0x80021000
E/LD:  Call stack:
E/LD:   0x800287a8
E/LD:   0x80021888
E/LD:   0x80029c00
E/LD:   0x80022108
TEEC_InvokeCommand(): 0xffff3024 (orig=3)
```